### PR TITLE
Add GH action script

### PR DESCRIPTION
**Pull request by veidenberg**
## Description
This PR will test a Github action script for syncing deleted branches from `ensembl-client` Github repo to its Gitlab mirror.
After merging this PR to the default branch (`dev`), all feature branches will run the script when the branch is deleted, issuing an API call to `enesmbl-client` GitLab repo for deleting the same branch from the GitLab mirror. This in turn will trigger the `Stop_review` job in GitLab CI/CD config, cleaning up the associated review app resources from kubernetes.

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/EWB-237

## Deployment URL(s)
All review apps

### CODE REVIEW

File: .github/workflows/delete_gitlab_branch.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Delete removed branch from GitLab mirror
+
+on:
+  delete:
+    branches-igore:
```
Comment (azangru) [inline .github/workflows/delete_gitlab_branch.yaml:5]: Igor? This guy?

Comment (veidenberg) [reply .github/workflows/delete_gitlab_branch.yaml]: Good catch :) Fixed.

**Review by azangru — APPROVED** [review]
I suppose we can't really test it until we merge it, right? 😄

**Comment by veidenberg** [comment]
Branch was deleted and its [review app cleanup job](https://gitlab.ebi.ac.uk/ensembl-web/ensembl-client/-/jobs/975512) was triggered. The job had trouble deleting the deployment part, but that is not related to this PR (same happened on tools api when triggering cleanup jobs manually).

**Pull request merged**